### DISCUSSION TITLE
Redirect to provider login page

### DISCUSSION THREAD

**Discussion by PenT1x** [nextauthjs/next-auth#8787] [Help]
### Summary

I use sveltekit and have a hooks.server.ts handle where I check protected routes. I only have one provider and therefore want to send the user directly to discords login page. 
I have tried using signIn('discord') but that gives this error:
```
const { callbackUrl = window.location.href, redirect = true } = options ?? {};
                                  ^
ReferenceError: window is not defined
```
How would you do this?

**Comment by PenT1x** [comment]
I have also tried
```
signIn('discord', { callbackUrl: event.url.href })
```
But then I get this error:
```
Error: Cannot use relative URL (/auth/csrf) with global fetch — use `event.fetch` instead: https://kit.svelte.dev/docs/web-standards#fetch-apis
```

**Reply by Zuntie** [reply]
I have no experience with Sveltekit, but that shouldn't matter much regarding this specific case.

In the next-auth configuration file i specified a specific "signIn" route (/login for example), and of course Discord as the provider.
My /login route checks if the user is authenticated. If not, it uses the **signIn("discord")** function. Therefore, i don't specify the callbackUrl.

This worked for me using NextJS.

I hope it could help.

**Reply by PenT1x** [reply]
I had totally forgotten about this. I have ended up doing the same. So in my hooks.server.ts I do this
```ts
if (event.url.pathname.startsWith("/dashboard")) {
        const session = await event.locals.getSession()
        
        if (!session || !session?.user) {
            throw redirect(303, "/login")
        }
    }

    return resolve(event)
```
